src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs

```diff
@@ -132,6 +132,9 @@ public class WorkItemJob : IQueueJob<WorkItemData>, IHaveLogger {
 
                 return JobResult.Success;
             } catch (Exception ex) {
+
+                await ReportProgressAsync(handler, queueEntry, -1, $"Failed: {ex.Message}").AnyContext();
```

Comment:
```suggestion
            if (queueEntry.Value.SendProgressReports)
                await ReportProgressAsync(handler, queueEntry, -1, ex.Message).AnyContext();
```

We should probably respect the `SendProgressReports` property on the queue item. I just noticed there is another spot where we aren't looking at this `SendProgressReports` flag in the `progressCallback `.

@ejsmith any concerns around try catching this here? I don't really like nesting this but if this fails, we won't abandon or release locks.

Reply:
We should respect the setting for sure. Try catch can't hurt.

Reply:
Added flag `SendProgressReports`